catowser/catowser/Coordinators/WebContentCoordinator.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  WebContentCoordinator.swift
+//  catowser
+//
+//  Created by Andrei Ermoshin on 11/21/22.
+//  Copyright © 2022 andreiermoshin. All rights reserved.
+//
+
+import UIKit
+import CoreHttpKit
+import CoreBrowser
+
+protocol WebContentDelegate: AnyObject {
+    func didProvisionalNavigationStart()
+    func didLoadingProgressChange(_ progress: Float)
+    func didProgress(show: Bool)
+}
+
+final class WebContentCoordinator: Coordinator {
+    let vcFactory: ViewControllerFactory
+    var startedCoordinator: Coordinator?
+    weak var parent: CoordinatorOwner?
+    var startedVC: AnyViewController?
+    weak var presenterVC: AnyViewController?
+
+    private let site: Site
+    private let jsPluginsSource: any JSPluginsSource
+    private let contentContainerView: UIView
+    
+    private var siteNavigationDelegate: SiteNavigationChangable? {
+        if UIDevice.current.userInterfaceIdiom == .phone {
+            return vcFactory.createdToolbaViewController as? SiteNavigationChangable
+        } else {
+            return vcFactory.createdDeviceSpecificSearchBarVC as? SiteNavigationChangable
+        }
+    }
+    private weak var delegate: WebContentDelegate?
+    
+    init(_ vcFactory: ViewControllerFactory,
+         _ presenter: AnyViewController,
+         _ contentContainerView: UIView,
+         _ delegate: WebContentDelegate,
+         _ site: Site,
+         _ jsPluginsSource: any JSPluginsSource) {
+        self.vcFactory = vcFactory
+        self.presenterVC = presenter
+        self.contentContainerView = contentContainerView
+        self.site = site
+        self.jsPluginsSource = jsPluginsSource
+        self.delegate = delegate
+    }
+    
+    func start() {
+        let webViewController = try? WebViewsEnvironment.shared.reuseManager.controllerFor(site, jsPluginsSource, self)
+        guard let vc = webViewController else {
+            assertionFailure("Failed create new web view for tab")
+            return
+        }
+        startedVC = vc
+        presenterVC?.viewController.add(asChildViewController: vc.viewController, to: contentContainerView)
+        
+        let topSitesView: UIView = vc.controllerView
+        topSitesView.translatesAutoresizingMaskIntoConstraints = false
+        topSitesView.leadingAnchor.constraint(equalTo: contentContainerView.leadingAnchor).isActive = true
+        topSitesView.trailingAnchor.constraint(equalTo: contentContainerView.trailingAnchor).isActive = true
+        topSitesView.topAnchor.constraint(equalTo: contentContainerView.topAnchor).isActive = true
+        topSitesView.bottomAnchor.constraint(equalTo: contentContainerView.bottomAnchor).isActive = true
+    }
+}
+
+enum WebContentRoute: Route {}
+
+extension WebContentCoordinator: Navigating {
+    typealias R = WebContentRoute
+    
+    func showNext(_ route: R) {}
```

Comment:
could be great to implement deep linking in this handler. The one which is about opening the installed app instead of loading the web site for the specific URLs.
See: `isNativeAppRedirectNeeded` in `WebViewModelImpl`